integration-cli/docker_api_containers_test.go

```diff
@@ -841,3 +841,22 @@ func (s *DockerSuite) TestStartWithTooLowMemoryLimit(c *check.C) {
 	c.Assert(status, check.Equals, http.StatusInternalServerError)
 	c.Assert(strings.Contains(string(b), "Minimum memory limit allowed is 4MB"), check.Equals, true)
 }
+
+func (s *DockerSuite) TestContainerApiRename(c *check.C) {
+	runCmd := exec.Command(dockerBinary, "run", "--name", "first_name", "-d", "busybox", "sh")
+	out, _, err := runCommandWithOutput(runCmd)
+	c.Assert(err, check.IsNil)
+
+	containerID := strings.TrimSpace(out)
+	newName := "new_name" + stringid.GenerateRandomID()
+	statusCode, _, err := sockRequestRaw("POST", "/containers/"+containerID+"/rename?name="+newName, nil, "")
```

Comment:
sockRequest will call sockRequestRaw and set the "application/json", since we don't pass arguments to it, I think it's fine to use sockRequest :)